app/web_form_handlers.py

```diff
@@ -0,0 +1,184 @@
+                if display_region == 'cy':
+                    # TODO Add Welsh Translation
+                    flash(request, WEBFORM_MISSING_EMAIL_EMPTY_MSG_CY)
+                else:
+                    flash(request, WEBFORM_MISSING_EMAIL_EMPTY_MSG)
+                form_valid = False
+
+            elif not WebForm.email_validation_pattern.fullmatch(str(data.get('email'))):
+                if display_region == 'cy':
+                    # TODO: Add Welsh Translation
+                    flash(request, WEBFORM_MISSING_EMAIL_INVALID_MSG_CY)
+                else:
+                    flash(request, WEBFORM_MISSING_EMAIL_INVALID_MSG)
+
+                form_valid = False
+
+        return form_valid
+
+
+@web_form_routes.view(r'/' + View.valid_display_regions + '/web-form/')
+class WebForm(WebForm):
+    @aiohttp_jinja2.template('web-form.html')
+    async def get(self, request):
+        self.setup_request(request)
+        display_region = request.match_info['display_region']
+        if display_region == 'cy':
+            # TODO Add Welsh Translation
+            page_title = 'Web Form'
+            locale = 'cy'
+        else:
+            page_title = 'Web Form'
+            locale = 'en'
+        self.log_entry(request, display_region + '/web-form')
+
+        return {
+            'display_region': display_region,
+            'page_title': page_title,
+            'locale': locale,
+            'page_url': View.gen_page_url(request)
+        }
+
+    @aiohttp_jinja2.template('web-form.html')
+    async def post(self, request):
+        self.setup_request(request)
+        display_region = request.match_info['display_region']
+        if display_region == 'cy':
+            # TODO Add Welsh Translation
+            page_title = 'Web Form'
+            locale = 'cy'
+        else:
+            page_title = 'Web Form'
+            locale = 'en'
+        self.log_entry(request, display_region + '/web-form')
+
+        data = await request.post()
+
+        form_valid = self.validate_form(request, data, display_region)
+
+        if not form_valid:
+            logger.info('web form submission error', client_ip=request['client_ip'])
+            return {
+                'form_value_name': data.get('name'),
+                'form_value_country': data.get('country'),
+                'form_value_query': data.get('query'),
+                'form_value_email': data.get('email'),
+                'form_value_description': data.get('description'),
+                'display_region': display_region,
+                'page_title': page_title,
+                'locale': locale,
+                'page_url': View.gen_page_url(request)
+            }
+
+        else:
+            logger.info('call web form endpoint', client_ip=request['client_ip'])
+            if display_region == 'cy':
+                language = 'CY'
+            else:
+                language = 'EN'
+            form_data = {
+                'category': data.get('query'),
```

Comment:
seems odd that this is not  `data.get('category')`

Reply:
Changed